### DISCUSSION TITLE
How to add padding

### DISCUSSION THREAD

**Discussion by thistlillo** [has2k1/plotnine#895] [Q&A]
I have the following code for creating an histogram (I cannot share the data).

```
hist = p9.ggplot(pd.DataFrame(label_frequency, columns=['frequency']).reset_index(), p9.aes(x='index', y='frequency')) + p9.geom_bar(stat='identity')
hist = hist + p9.theme_minimal() + p9.theme(aspect_ratio=3/4) + p9.ggtitle("Counts")
hist = hist + p9.xlab("name") + p9.ylab("count")
hist = hist + p9.geom_text(p9.aes(label='frequency'), position=p9.position_dodge(width=0.9), va='bottom')
hist = hist + p9.theme(panel_border=p9.element_rect(color='gray', linetype='solid', fill='none'))
hist = hist + p9.theme(axis_text_x=p9.element_blank())
```
and I get this:
![Screenshot 2024-12-05 at 14 36 23](https://github.com/user-attachments/assets/d1f300e0-25ce-4310-9d94-80defee778f8)

Now I would like to add some padding on top of the chart, to have some empty space between the highest number 264 and the border.
 
I am not able to do it.

I have found a parameter `plot_margin_top`. When I use it I get a bad image and I am not sure that it corresponds to the padding.

Can you please help me?

**Comment by has2k1** [comment]
You need to increase the scale expansion.

```python
scale_y_continuous(expand=(0.05, 0, 0.05, 0)) # default values, expand the bottom & top by 5%
scale_y_continuous(expand=(0, 0, 0.075, 0)) # expand the top by 7.5%
scale_y_continuous(expand=(0, 10, 0.075, 0)) # expand the bottom by 10, expand the top by 7.5%
```
